CDCgov/cdc-open-viz
# Dev 7020 markup include

Testing Scenario setup:

- User is in COVE Editor
- Select Dashboard Visualization
- Select any data source
- Drag and drop the markup include to the visualization panel
- Either configure the data source with the gear icon or do not use any data source. 
- Click to tool icon to bring up the table editor panel.

Testing instructions:

1. Edit the Header
- Open the General Accordion on the left.
- The header title can be changed from the text input.
- If the Header is left blank, then the header section will not be shown in the Markup Include.

2. Add HTML Markup:
- Open the Content Editor accordion on the left.
- The "USE INLINE HTML" checkbox with toggle between using the HTML markup and the legacy URL markup.
- Within the HTML text box, adding basic HTML elements will show up on the Markup Include section.
- Bootstrap classes and inline styling will work in this section.
- Variables can be added here using the following pattern {{variable name}}.
   - example: Variable name of State would be written out as {{State}}.
   - see next step for adding Variables.
  
3. Adding variables
- To use Variables, a data source needs to be imported and configured.
- Below the HTML Text Box, there is the Variable section.
- Click the "Create New Variable" button and a "New Variable" with a down arrow button will appear.
- Clicking on the arrow will open up the new variable.
- Add a name to the "VARIABLE NAME" input. The name can be renamed later if needed.
- Once the name is inputted, the "COLUMN" dropdown with be available.
- Select a Column. This will be the output of the variable.
  - example: If there is a State column, the variable will display the state that meets the conditions set by global page filter and the optional conditions that will be set below.

4. Setting conditions 
- After the variable name and column are selected, the "ADD CONDITION" button will be available. 
- Conditions are optional and can add as many as needed.
- Click the "Add Condition" button and a "NEW CONDITION" with a down arrow button will appear.
- Clicking on the arrow will open up the new condition.
- The first select will set the condition to validate
- The second select will set if the condition is or is not equal to the value
- The third select will set the value of the condition
- After the "Add Condition" can be pushed again or "Done" if no more conditions are needed

5. Editing and deleting variables
- Click on the down arrow of the Variable that needs to be edited or deleted
- To edit the variable, just change any field that was set. 
- If "COLUMN" is changed, this will delete the conditions that were set earlier for that variable because conditions are based off the column. 
- Press the "Delete" button to delete the variable

6. Editing and deleting conditions
- Click on the down arrow of the Condition that needs to be edited or deleted
- To edit the condition, just change any select that was set. 
- If the first select is changed, this will reset the third select  
- Press the "Remove" button to delete the condition

7. Using variables
- As stated in step 2, to add a variable to the Markup Include, wrap the variable in {{variable name}}.
- When a variable is used, in the Markup, the variable will be highlighted yellow. 
- Hover over the variable to see the possible values that will replace the variable on a live page.
- These are possible values that will be determine by the global filter on the page.
- The same variable can be added to the HTML as many times as needed.
